packages/components/src/date-time/date/index.tsx

```diff
@@ -69,14 +64,13 @@ function DatePickerDay( { day, events = [] }: DatePickerDayProps ) {
 	}, [ events.length ] );
 
 	return (
-		<div
+		<Day
 			ref={ ref }
-			className={ classnames( 'components-datetime__date__day', {
-				'has-events': events?.length,
-			} ) }
+			className="components-datetime__date__day"
```

Comment:
We're no longer using most of the class names but I've tried to keep them for backwards compatibility. I don't think this is a hard requirement, but maybe nice to do anyway.